cpp/src/strings/split/split.cu

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/strings/split/split.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/utilities/error.hpp>
+#include <strings/utilities.hpp>
+
+#include <vector>
+#include <thrust/transform.h>
+
+namespace cudf
+{
+namespace strings
+{
+namespace detail
+{
+
+using string_index_pair = thrust::pair<const char*,size_type>;
+
+namespace
+{
+
+/**
+ * @brief Common token counter for all split methods in this file.
+ */
+struct token_counter_fn
+{
+    column_device_view const d_strings;
+    string_view const d_delimiter;
+    size_type tokens;
+
+    // returns the number of possible tokens in each string
+    __device__ size_type operator()(size_type idx) const
```

Comment:
> Using a pair_iterator could allow this operator to work on a <string_view, is_null> pair rather than an index...

Is this change required to approve this PR?

Reply:
No.